lib/rubocop/cop/rails/time_zone.rb

```diff
@@ -228,12 +228,25 @@ def acceptable_methods(klass, method_name, node)
           acceptable
         end
 
-        # Time.new can be called with a time zone offset
+        # Time.new, Time.at, and Time.now can be called with a time zone offset
         # When it is, that should be considered safe
         # Example:
         # Time.new(1988, 3, 15, 3, 0, 0, "-05:00")
         def offset_provided?(node)
-          node.arguments.size >= 7
+          case node.method_name
+          when :new
+            node.arguments.size >= 7 || offset_option_provided?(node)
```

Comment:
`Time.new` accepts up to 7 arguments. Not more than 7.

```ruby
Time.new(2022, 2, 3, 4, 5, 6, '-05:00')    #=> 2022-02-03 04:05:06 -0500
Time.new(2022, 2, 3, 4, 5, 6, 7, '-05:00') #=> :in `initialize': wrong number of arguments (given 8, expected 1..7) (ArgumentError)
```

This intention like this? It seems that the original logic was may not good.

```suggestion
            node.arguments.size == 7 || offset_option_provided?(node)
```